cpu/atmega_common/periph/gpio.c

```diff
@@ -416,6 +416,32 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     return 0;
 }
 
+int gpio_update_cb(gpio_t pin, gpio_cb_t cb)
+{
+    int8_t int_num = _int_num(pin);
+
+    if (int_num < 0) {
+        return -1;
+    }
+
+    _gpio_update_cb(pin, &config[int_num].cb, cb);
+
+    return 0;
+}
+
+int gpio_update_cb_arg(gpio_t pin, void *arg)
+{
+    int8_t int_num = _int_num(pin);
+
+    if (int_num < 0) {
+        return -1;
+    }
```

Comment:
One ATmega specific detail: The ATmegas only have individual interrupt vectors for about 3 pins, but shared interrupt vectors for a group of 8 pins (called pin changed interrupts). On top of the pin changed interrupts a RIOT compatible `gpio_init_int()` is implemented, when pseudomodule is used. `_int_num()` will however only work for "proper" interrupts, not for pin changed interrupts.

(But I'd say keep it like this for now. I'd even say that it is fine to merge the PR without support for this special case in order to not get lost in the implementation details of a dying platform. This can be easily added later on.)

Reply:
Atmega is the C64 of the MCU world, I think it will be with us for a bit.
I can just move the code from `gpio_init_int()` into a helper function or call the `gpio_update_cb()`  function from there.

Reply:
> Atmega is the C64 of the MCU world

:laughing: :+1: 

And there is actually development on that platform. (Not from Microchip, but there is an [improved clone](https://www.avrfreaks.net/sites/default/files/forum_attachments/LGT8F88P%20LGT8F168P%20LGT8F328P%20translated.pdf) of the ATmega328P - seemingly with a proper on chip debug interface (instead of debugWIRE)